### PR TITLE
Remove run_jedi_exe and cleanup legacy YAML tools

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
We need to switch to using the `wxflow` jinja-based YAML templating tools everywhere. This PR removes all (that I can see) of the old YAML tool usage, as well as the `run_jedi_exe` script.

### CODE REVIEW

**Review by jiaruidong2017 — DISMISSED** [review]
Please remove `ush/land/genYAML_output_letkfoi.yaml`.

**Review by RussTreadon-NOAA — DISMISSED** [review]
ATM 3dvar, 3denvar, and letkf changes (deletions) look good to me.

Approve.

File: scripts/exgdas_global_marine_analysis_prep.py [hunk 1]
```diff
@@ -517,22 +517,17 @@ def find_clim_ens(input_date):
...
 for varchgyaml in varchgyamls:
-    soca2cice_cfg['template'] = os.path.join(gdas_home, 'parm', 'soca', 'varchange', varchgyaml)
-    f = open('tmp.yaml', 'w')
-    # TODO: use YAMLFile instead
-    yaml.dump(soca2cice_cfg, f, sort_keys=False, default_flow_style=False)
-    ufsda.genYAML.genYAML('tmp.yaml', output=varchgyaml)
+    soca2cice_cfg_template = os.path.join(gdas_home, 'parm', 'soca', 'varchange', varchgyaml)
+    outyaml = YAMLFile(path=soca2cice_cfg_template)
+    outyaml = Template.substitute_structure(outyaml, TemplateConstants.DOLLAR_PARENTHESES, soca2cice_cfg.get)
+    outyaml.save(varchgyaml)
```
Comment (guillaumevernieres) [inline scripts/exgdas_global_marine_analysis_prep.py:530]: Thanks for doing my TODO's @CoryMartin-NOAA !

**Review by guillaumevernieres — DISMISSED** [review]

**Comment by CoryMartin-NOAA** [comment]
I removed the extraneous file that @jiaruidong2017 suggested, and now need a re-approve from folks!

**Review by jiaruidong2017 — APPROVED** [review]

**Review by RussTreadon-NOAA — APPROVED** [review]
Approve

**Pull request merged**